manticore/manticore.py

```diff
@@ -76,15 +76,15 @@ def make_decree(program, concrete_start='', **kwargs):
     return initial_state
 
 
-def make_linux(program, argv=None, env=None, symbolic_files=None, concrete_start=''):
+def make_linux(program, argv=None, env=None, entry=None, symbolic_files=None, concrete_start=''):
     env = {} if env is None else env
     argv = [] if argv is None else argv
     env = ['%s=%s' % (k, v) for k, v in env.items()]
 
     logger.info('Loading program %s', program)
 
     constraints = ConstraintSet()
-    platform = linux.SLinux(program, argv=argv, envp=env,
+    platform = linux.SLinux(program, argv=argv, envp=env, entry=entry,
```

Comment:
I've been looking over this code and trying to determine why this feels like it makes `SLinux` API feel heavy. I am not sure that modifying the loading process to add an entry is the cleanest way to go; otherwise if we add future support for other methods of initialization, it'll grow it further. After some thought, what if we do the following:

Leave the loading process as it was, and add a `set_entry()` method that takes a virtual address.[1] You can then use that and `_find_symbol()` below (or potentially renamed to `_resolve_symbol()`) to alter the entry point after initialization. This way, we also won't introduce ambiguous parameters (`entry`/`entrysymbol`/etc) to the constructor and our elf loading process more closely resembles the actual process in Linux. 

So new initialization will roughly look like:
```python
platform = linux.SLinux(platform, argv, etc...)
if args.entrysymbol and args.entry:
    # raise an exception and notify that they should be mutually exclusive
if args.entrysymbol:
    entry = platform._find_symbol(args.entrysymbol)
    platform.set_entry(entry)
if args.entry:
    platform.set_entry(args.entry)
initial_state = State(constraints, platform)
...
etc
```

What are your thoughts?

[1] We might be able to extend this API to other platforms.

Reply:
Looks good. I just had to store the load_addr variable so that it an be used in `set_entry`